### ISSUE TITLE
Client: Release-blocking Client Bug

### ISSUE THREAD

**Issue by holgerd77** [ethereumjs/ethereumjs-monorepo#1930]
We have a too severe bug in the client to continue with the releases.

So basically: if a chain has a partially synced state (or is just synced from the beginning) and then the client is re-started we repeatedly get the following error:

![grafik](https://user-images.githubusercontent.com/931137/171608426-3c2aadf6-6c60-4539-b980-ac6510e2d920.png)

(here is the initial sync for comparison, going well:)

![grafik](https://user-images.githubusercontent.com/931137/171608531-2f0a8f79-19c3-47db-bda2-59419bdb44ce.png)

I can confirm this for at least Rinkeby and Sepolia, for Mainnet things seem to work (so maybe this is a PoA related thing?).

This bug is *urgent* since it is actively blocking the current release round.

**Comment by holgerd77** [comment]
Can confirm that this is a Clique-related bug.

So "recently signed" is thrown along block addition [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/blockchain/src/index.ts#L944) in Blockchain, this calls into the [cliqueCheckRecentlySigned()](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/blockchain/src/index.ts#L472) method for this error to be triggered.

So I guess it can very likely be a starter thing, that this comparison gets evaluated wrong or towards a wrong initialized state along Blockchain initialization.

Any idea what change might have caused this respectively had there been work in these code parts (Blockchain) lately? Or in related code parts in the client? 🤔

**Comment by holgerd77** [comment]
Ok, getting a bit closer:

For a Rinkeby (re-)run I am getting the following for `this._cliqueLatestBlockSigners` captured in `cliqueCheckRecentlySigned()`:

```shell
[
  [
    <BN: bea>,
    Address {
      buf: <Buffer b2 79 18 2d 99 e6 57 03 f0 07 6e 48 12 65 3a ab 85 fc a0 f0>
    }
  ],
  [
    <BN: b87>,
    Address {
      buf: <Buffer b2 79 18 2d 99 e6 57 03 f0 07 6e 48 12 65 3a ab 85 fc a0 f0>
    }
  ]
]
Limit:2, Signers:2, Seen:2
```

So this is the same signer, however the block numbers are far apart (bea ~ 3500 (this is my current sync state) and b87 ~ 2951). I guess this shouldn't be the case and block numbers in this data structure should be consecutive to be meaningful if I get this right.

**Comment by holgerd77** [comment]
Here is a stack trace for the call context:

```shell
Error
    at Blockchain.cliqueCheckRecentlySigned (/ethereumjs-monorepo/packages/blockchain/src/index.ts:483:15)
    at /ethereumjs-monorepo/packages/blockchain/src/index.ts:947:20
    at async Blockchain.runWithLock (/ethereumjs-monorepo/packages/blockchain/src/index.ts:453:21)
    at async Blockchain._putBlockOrHeader (/ethereumjs-monorepo/packages/blockchain/src/index.ts:902:5)
    at async Blockchain.putBlock (/ethereumjs-monorepo/packages/blockchain/src/index.ts:857:5)
    at async Chain.putBlocks (/ethereumjs-monorepo/packages/client/lib/blockchain/chain.ts:293:7)
    at async BlockFetcher.store (/ethereumjs-monorepo/packages/client/lib/sync/fetcher/blockfetcher.ts:103:19)
    at async Writable._write (/ethereumjs-monorepo/packages/client/lib/sync/fetcher/fetcher.ts:393:11)
```

**Comment by holgerd77** [comment]
Possible fix in #1931

**Comment by Dayday10** [comment]
> We have a too severe bug in the client to continue with the releases.
> 
> So basically: if a chain has a partially synced state (or is just synced from the beginning) and then the client is re-started we repeatedly get the following error:
> 
> ![grafik](https://user-images.githubusercontent.com/931137/171608426-3c2aadf6-6c60-4539-b980-ac6510e2d920.png)
> 
> (here is the initial sync for comparison, going well:)
> 
> ![grafik](https://user-images.githubusercontent.com/931137/171608531-2f0a8f79-19c3-47db-bda2-59419bdb44ce.png)
> 
> I can confirm this for at least Rinkeby and Sepolia, for Mainnet things seem to work (so maybe this is a PoA related thing?).
> 
> This bug is *urgent* since it is actively blocking the current release round.
> 
>
